src/tests/test_sexual_behaviour.py

```diff
@@ -52,7 +57,25 @@ def check_num_partners(row):
 def test_num_partners():
     """Check that number of partners are reasonable"""
     pop_data = Population(size=1000, start_date=date(1989, 1, 1)).data
-    print(pop_data)
+    assert(any(pop_data["num_partners"] > 0))
     # Check the num_partners column
     checks = pop_data.apply(check_num_partners, axis=1)
     assert np.all(checks)
+
+
+def test_sex_behaviour_groupings():
+    """Check that people are assigned to all sex behaviour groups!"""
+    pop_data = Population(size=1000, start_date=date(1989, 1, 1)).data
+    for sex in SexType:
+        for group in SexBehaviours[sex]:
+            index = selector(pop_data, sex=(operator.eq, sex), sex_behaviour=(operator.eq, group))
+            assert any(index)
```

Comment:
This is fine, although one suggestion to also check that we are not assigning values outside the expected (I think in that case pandas will silently change the type of the column instead of complaining, but I may be wrong):
```suggestion
    for sex in SexType:
        groups_in_data = pop_data[pop_data.sex == sex].sex_behaviour.unique()
        assert sorted(groups_in_data) == sorted(SexBehaviours[sex])
```